### PR TITLE
Extract lib audio io

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: #1136 

GUI Toolkit-neutral library for the audio engine; and a subset of that, just for the realtime effect management, can
be turned into another small library that it depends on.

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Comment by Paul-Licameli** [comment]
A force push only to add, at the proper commit in the sequence, a missing #include

**Comment by Paul-Licameli** [comment]
Sorry for the force push.  Only minimal changes to rebase onto latest master, resolve a simple conflict in include directories of AudioIO.cpp, and fix the Windows link with visibility of the structure `RealtimeEffectManager::AllListsLock`

File: src/AudioIO.h [hunk 1]
```diff
@@ -76,12 +80,12 @@ struct AudioIOEvent {
 };
 
 struct TransportTracks {
-   WaveTrackArray playbackTracks;
-   WaveTrackArray captureTracks;
+   WritableSampleTrackArray playbackTracks;
```
Comment (crsib) [inline src/AudioIO.h]: I wonder why playback tracks should be writable

**Review by crsib — APPROVED** [review]
A very welcome PR! Please fix the link failure before merging.

File: src/AudioIO.h (hunk 1 above)
Comment (Paul-Licameli) [reply src/AudioIO.h]: Later commit addressed that, yes?

Comment (crsib) [reply src/AudioIO.h]: Yes!

**Pull request merged**